instance/tasks.py

```diff
@@ -59,21 +59,19 @@ def spawn_appserver(instance_ref_id, mark_active_on_success=False, num_attempts=
         appserver_id = instance.spawn_appserver()
         if appserver_id:
             if mark_active_on_success:
-                # If the AppServer provisioned successfully, make it the active one:
-                # Note: if I call spawn_appserver() twice, and the second one provisions sooner, the first one may then
-                # finish and replace the second as the active server. We are not really worried about that for now.
-                instance.set_appserver_active(appserver_id)
+                # If the AppServer provisioned successfully, make it active.
+                appserver_make_active(appserver_id)
```

Comment:
@pomegranited You are right, for the first appserver of an instance this still makes sense.